src/main/java/com/google/devtools/build/lib/packages/TargetUtils.java

```diff
@@ -234,6 +231,23 @@ private static boolean hasConstraint(Rule rule, String keyword) {
     return ImmutableMap.copyOf(map);
   }
 
+  /**
+   * Returns the execution info from the tags declared on the target. These include only some tags
+   * {@link #legalExecInfoKeys} as keys with empty values.
+   *
+   * @param rule a rule instance to get tags from
+   * @param allowTagsPropagation if set to true, tags will be propagated from a target to the
+   *        actions' execution requirements, for more details {@see
+   *        SkylarkSematicOptions#experimentalAllowTagsPropagation}
+   */
+  public static Map<String, String> getExecutionInfo(Rule rule, boolean allowTagsPropagation) {
```

Comment:
this method is only called in three different sides. Remove it and instead use ternary operator at the call side?

Reply:
done